File: scanner/scanner/models/studio.py
from __future__ import annotations

from ..utils import Model
from .metadataid import MetadataId


class Studio(Model):
	slug: str
	external_id: dict[str, MetadataId]
	translations: dict[str, StudioTranslation] = {}


class StudioTranslation(Model):
	name: str
	logo: str | None

File: front/src/models/studio.ts
import { z } from "zod/v4";
import { KImage } from "./utils/images";
import { Metadata } from "./utils/metadata";
import { zdate } from "./utils/utils";

export const Studio = z.object({
	id: z.string(),
	slug: z.string(),
	name: z.string(),
	logo: KImage.nullable(),
	externalId: Metadata,
	createdAt: zdate(),
	updatedAt: zdate(),
});
export type Studio = z.infer<typeof Studio>;
